shipyard_blueprints/email/shipyard_email/cli/send_email_with_file.py

```diff
@@ -0,0 +1,155 @@
+    parser.add_argument("--cc", dest="cc", default="", required=False)
+    parser.add_argument("--bcc", dest="bcc", default="", required=False)
+    parser.add_argument("--username", dest="username", default="", required=False)
+    parser.add_argument("--password", dest="password", default="", required=True)
+    parser.add_argument("--subject", dest="subject", default="", required=False)
+    parser.add_argument("--message", dest="message", default="", required=True)
+    parser.add_argument(
+        "--source-file-name", dest="source_file_name", default="", required=False
+    )
+    parser.add_argument(
+        "--source-folder-name", dest="source_folder_name", default="", required=False
+    )
+    parser.add_argument(
+        "--source-file-name-match-type",
+        dest="source_file_name_match_type",
+        default="exact_match",
+        choices={"exact_match", "regex_match"},
+        required=False,
+    )
+    # TODO: Remove unused arguments when blueprints can be updated safely
+
+    parser.add_argument(
+        "--conditional-send",  # Left in for backwards compatibility
+        dest="conditional_send",
+        default="always",
+        required=False,
+        choices={"file_exists", "file_dne", "always"},
+    )
+
+    parser.add_argument(
+        "--file-upload",  # Left in for backwards compatibility
+        dest="file_upload",
+        default="yes",
+        required=False,
+    )
+    parser.add_argument(
+        "--include-shipyard-footer",
+        dest="include_shipyard_footer",
+        default="TRUE",
+        choices={"TRUE", "FALSE"},
+        required=False,
+    )
+
+    args = parser.parse_args()
+    if not (args.to or args.cc or args.bcc):
+        raise InvalidInputError(
+            "Email requires at least one recipient using --to, --cc, or --bcc"
+        )
+    return args
+
+
+def main():
+    try:
+        args = get_args()
+        send_method = args.send_method.lower() or "tls"
+        sender_address = args.sender_address
+        username = args.username
+        message = args.message
+
+        if args.include_shipyard_footer:
+            include_shipyard_footer = shipyard.args.convert_to_boolean(
+                args.include_shipyard_footer
+            )
+        else:
+            logger.warning("include_shipyard_footer not set. Defaulting to TRUE.")
+            include_shipyard_footer = True
+
+        if not username:
+            username = sender_address
+
+        client = EmailClient(
+            args.smtp_host, args.smtp_port, username, args.password, send_method
+        )
+
+        source_file_name = args.source_file_name
+        source_folder_name = shipyard.files.clean_folder_name(args.source_folder_name)
+        file_paths = shipyard.files.find_matching_files(
+            source_file_name, source_folder_name, args.source_file_name_match_type
+        )
+
+        if not file_paths:
+            raise InvalidFileInputError(
```

Comment:
Will this be caught in the `raise ExitCodeException` part? If not, then that needs to be handled and exited

Reply:
it would be caught with `ExitCodeException` it is an exception derived from that class